File: Seminarji/seminar2/Java/Seminar2/src/Seminar2.java
import java.awt.image.AreaAveragingScaleFilter;
import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Seminar2 {
    public static void main(String args[])
    {
        String file_name = args[0];

        try (Scanner in = new Scanner(new File(file_name))){
            int numberTestCases = Integer.parseInt(in.nextLine().stripTrailing());
            final int X = 1;
            final int Y = 0;

            for (int i = 1; i <= numberTestCases; i++) {
                List<Integer> list = new ArrayList<>();
                System.out.printf("Case %d:\n", i);
                int M = Integer.parseInt(in.nextLine().stripTrailing());
                StringBuilder piratesM = new StringBuilder();
                for (int j = 0; j < M; j++) {
                    int T = Integer.parseInt(in.nextLine().stripTrailing());
                    String piratesT = in.nextLine().stripTrailing().repeat(T);
                    piratesM.append(piratesT);
                }

                String[] integerStrings = piratesM.toString().split("");
                // Splits each spaced integer into a String array.
                for (int z = 0; z < integerStrings.length; z++){
                    list.add(Integer.parseInt(integerStrings[z]));
                }


                int Q = Integer.parseInt(in.nextLine().stripTrailing());
                int outputNum = 1;
                for (int j = 0; j < Q; j++) {
                    char c;
                    int a,b;
                    String tempArray[] = in.nextLine().stripTrailing().split(" ");
                    c = tempArray[0].charAt(0);
                    a = Integer.parseInt(tempArray[1]);
                    b = Integer.parseInt(tempArray[2]);


                    switch (c){
                        case 'F':
                            for (int k = a; k <= b ; k++) {
                                list.set(k, X);
                            }
                            break;
                        case 'E':
                            for (int k = a; k <= b ; k++) {
                                list.set(k, Y);
                            }
                            break;
                        case 'I':
                            for (int k = a; k <= b; k++) {
                                list.set(k, list.get(k) ^ 1);
                            }
                            break;
                        case 'S':
                            int sum = 0;
                            for (int k = a; k <= b ; k++) {
                                sum += list.get(k);
                            }
                            System.out.printf("Q%d: %d\n" , outputNum++, sum);
                            break;
                    }

                }


            }

        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
    }
}

File: Seminarji/seminar2/Seminar2/Seminar2.py
import sys


def main():
    input_file_name = sys.argv[1]
    input_file = open(input_file_name, "r")
    number_test_cases = int(input_file.readline().rstrip())
    X = 1
    Y = 0
    for i in range(1, number_test_cases + 1):
        print("Case %d:" % i)
        M = int(input_file.readline().rstrip())
        piratesM = ""
        for j in range(1, M + 1):
            T = int(input_file.readline().rstrip())
            piratesT = input_file.readline().rstrip()
            piratesT = piratesT * T
            piratesM += piratesT

        arr = [int(i) for i in str(piratesM)]
        n = len(arr)


        Q = int(input_file.readline())
        output_num = 1
        for j in range(1, Q + 1):
            char, a, b = input_file.readline().rstrip().split(" ")
            a = int(a)
            b = int(b)
            if char == 'F':
                for z in range(a, b + 1):
                    arr[z] = X
            elif char == 'E':
                for z in range(a, b + 1):
                    arr[z] = Y

            elif char == 'I':
                for z in range(a, b + 1):
                    arr[z] ^= 1

            elif char == 'S':
                sum = 0
                for z in range(a, b + 1):
                    sum += arr[z]
                print("Q%d: %d" % (output_num, sum))
                output_num += 1

    input_file.close()


if __name__ == "__main__":
    main()
